### ISSUE TITLE
TIMEOUT: src/php/bin/run_tests.sh Handshake failed with fatal error routines:OPENSSL_internal:WRONG_VERSION_NUMBER.

### ISSUE THREAD

**Issue by AspirinSJL** [grpc/grpc#14165]
https://sponge.corp.google.com/target?id=066dbe4a-6b74-434f-92e5-21477e816380&target=github/grpc/php_linux_opt_native&searchFor=&show=FAILED&sortBy=STATUS

```
Timeout
g test 'ChannelTest::testGetConnectivityStateWithBool'.
.
Starting test 'ChannelTest::testGetTarget'.
.
Starting test 'ChannelTest::testWatchConnectivityState'.
.
Starting test 'ChannelTest::testClose'.
.
Starting test 'ChannelTest::testInvalidConstructorWithNull'.
.
Starting test 'ChannelTest::testInvalidConstructorWith'.
.
Starting test 'ChannelTest::testInvalidCredentials'.
.
Starting test 'ChannelTest::testInvalidOptionsArray'.
.
Starting test 'ChannelTest::testInvalidGetConnectivityStateWithArray'.
.
Starting test 'ChannelTest::testInvalidWatchConnectivityState'.
.
Starting test 'ChannelTest::testInvalidWatchConnectivityState2'.
.
Starting test 'ChannelTest::testPersistentChannelSameHost'.
.
Starting test 'ChannelTest::testPersistentChannelDifferentHost'.
.
Starting test 'ChannelTest::testPersistentChannelSameArgs'.
.
Starting test 'ChannelTest::testPersistentChannelDifferentArgs'.
.
Starting test 'ChannelTest::testPersistentChannelSameChannelCredentials'.
E0124 01:15:52.348198913    8877 security_connector.cc:944]  load_file: {"created":"@1516756552.348182438","description":"Failed to load file","file":"src/core/lib/iomgr/load_file.cc","file_line":69,"filename":"/usr/local/share/grpc/roots.pem","referenced_errors":[{"created":"@1516756552.348181333","description":"OS Error","errno":2,"file":"src/core/lib/iomgr/load_file.cc","file_line":43,"os_error":"No such file or directory","syscall":"fopen"}]}
E0124 01:15:52.348219238    8877 security_connector.cc:990]  Could not get default pem root certs.
E0124 01:15:52.348223473    8877 secure_channel_create.cc:113] Failed to create secure subchannel for secure name 'localhost:1'
E0124 01:15:52.348226543    8877 secure_channel_create.cc:144] Failed to create subchannel arguments during subchannel creation.
E0124 01:15:52.348234373    8877 security_connector.cc:990]  Could not get default pem root certs.
E0124 01:15:52.348237273    8877 secure_channel_create.cc:113] Failed to create secure subchannel for secure name 'localhost:1'
E0124 01:15:52.348240300    8877 secure_channel_create.cc:144] Failed to create subchannel arguments during subchannel creation.
.
Starting test 'ChannelTest::testPersistentChannelDifferentChannelCredentials'.
E0124 01:15:52.348753492    8877 security_connector.cc:990]  Could not get default pem root certs.
E0124 01:15:52.348761625    8877 secure_channel_create.cc:113] Failed to create secure subchannel for secure name 'localhost:1'
E0124 01:15:52.348764426    8877 secure_channel_create.cc:144] Failed to create subchannel arguments during subchannel creation.
E0124 01:15:52.348770705    8877 security_connector.cc:990]  Could not get default pem root certs.
E0124 01:15:52.348773654    8877 secure_channel_create.cc:113] Failed to create secure subchannel for secure name 'localhost:1'
E0124 01:15:52.348776201    8877 secure_channel_create.cc:144] Failed to create subchannel arguments during subchannel creation.
.
Starting test 'ChannelTest::testPersistentChannelSameChannelCredentialsRootCerts'.
.
Starting test 'ChannelTest::testPersistentChannelDifferentSecureChannelCredentials'.
E0124 01:15:52.350109985    8877 security_connector.cc:990]  Could not get default pem root certs.
E0124 01:15:52.350120265    8877 secure_channel_create.cc:113] Failed to create secure subchannel for secure name 'localhost:1'
E0124 01:15:52.350123921    8877 secure_channel_create.cc:144] Failed to create subchannel arguments during subchannel creation.
E0124 01:15:52.350130033    8877 security_connector.cc:990]  Could not get default pem root certs.
E0124 01:15:52.350132896    8877 secure_channel_create.cc:113] Failed to create secure subchannel for secure name 'localhost:1'
E0124 01:15:52.350135580    8877 secure_channel_create.cc:144] Failed to create subchannel arguments during subchannel creation.
.
Starting test 'ChannelTest::testPersistentChannelSharedChannelClose'.
.
Starting test 'ChannelTest::testPersistentChannelCreateAfterClose'.
.
Starting test 'ChannelTest::testPersistentChannelSharedMoreThanTwo'.
.
Starting test 'ChannelTest::testPersistentChannelWithCallCredentials'.
E0124 01:15:52.351805366    8877 security_connector.cc:990]  Could not get default pem root certs.
E0124 01:15:52.351814290    8877 secure_channel_create.cc:113] Failed to create secure subchannel for secure name 'localhost:1'
E0124 01:15:52.351817210    8877 secure_channel_create.cc:144] Failed to create subchannel arguments during subchannel creation.
E0124 01:15:52.351823947    8877 security_connector.cc:990]  Could not get default pem root certs.
E0124 01:15:52.351826734    8877 secure_channel_create.cc:113] Failed to create secure subchannel for secure name 'localhost:1'
E0124 01:15:52.351829259    8877 secure_channel_create.cc:144] Failed to create subchannel arguments during subchannel creation.
.
Starting test 'ChannelTest::testPersistentChannelWithDifferentCallCredentials'.
E0124 01:15:52.352356965    8877 security_connector.cc:990]  Could not get default pem root certs.
E0124 01:15:52.352365412    8877 secure_channel_create.cc:113] Failed to create secure subchannel for secure name 'localhost:1'
E0124 01:15:52.352368103    8877 secure_channel_create.cc:144] Failed to create subchannel arguments during subchannel creation.
E0124 01:15:52.352374105    8877 security_connector.cc:990]  Could not get default pem root certs.
E0124 01:15:52.352376976    8877 secure_channel_create.cc:113] Failed to create secure subchannel for secure name 'localhost:1'
E0124 01:15:52.352379450    8877 secure_channel_create.cc:144] Failed to create subchannel arguments during subchannel creation.
.
Starting test 'ChannelTest::testPersistentChannelForceNew'.
.
Starting test 'ChannelTest::testPersistentChannelForceNewOldChannelIdle'.
.
Starting test 'ChannelTest::testPersistentChannelForceNewOldChannelClose'.
.
Starting test 'ChannelTest::testPersistentChannelForceNewNewChannelClose'.
.
Starting test 'EndToEndTest::testSimpleRequestBody'.
.
Starting test 'EndToEndTest::testMessageWriteFlags'.
.
Starting test 'EndToEndTest::testClientServerFullRequestResponse'.
.
Starting test 'EndToEndTest::testInvalidClientMessageArray'.
.  63 / 113 ( 55%)

Starting test 'EndToEndTest::testInvalidClientMessageString'.
.
Starting test 'EndToEndTest::testInvalidClientMessageFlags'.
.
Starting test 'EndToEndTest::testInvalidServerStatusMetadata'.
.
Starting test 'EndToEndTest::testInvalidServerStatusCode'.
.
Starting test 'EndToEndTest::testMissingServerStatusCode'.
.
Starting test 'EndToEndTest::testInvalidServerStatusDetails'.
.
Starting test 'EndToEndTest::testMissingServerStatusDetails'.
.
Starting test 'EndToEndTest::testInvalidStartBatchKey'.
.
Starting test 'EndToEndTest::testInvalidStartBatch'.
.
Starting test 'EndToEndTest::testGetTarget'.
.
Starting test 'EndToEndTest::testGetConnectivityState'.
.
Starting test 'EndToEndTest::testWatchConnectivityStateFailed'.
.
Starting test 'EndToEndTest::testWatchConnectivityStateSuccess'.
.
Starting test 'EndToEndTest::testWatchConnectivityStateDoNothing'.
.
Starting test 'EndToEndTest::testGetConnectivityStateInvalidParam'.
.
Starting test 'EndToEndTest::testWatchConnectivityStateInvalidParam'.
.
Starting test 'EndToEndTest::testChannelConstructorInvalidParam'.
.
Starting test 'EndToEndTest::testClose'.
.
Starting test 'SecureEndToEndTest::testSimpleRequestBody'.
.
Starting test 'SecureEndToEndTest::testMessageWriteFlags'.
.
Starting test 'SecureEndToEndTest::testClientServerFullRequestResponse'.
E0124 01:15:52.538554621    8876 ssl_transport_security.cc:978] Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
E0124 01:15:52.538847180    8876 ssl_transport_security.cc:978] Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
E0124 01:15:53.538792569    8876 ssl_transport_security.cc:978] Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
E0124 01:15:53.539149511    8876 ssl_transport_security.cc:978] Handshake failed with fatal error SSL_ERROR_SSL: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER.
```

**Comment by ZhouyihaiDing** [comment]
I haven't reproduced it running the same tests many time. Is it possible related to this issue? [#9538](https://github.com/grpc/grpc/issues/9538)

**Comment by AspirinSJL** [comment]
Looks similar. Although #9538 is a test failure other than timeout.

**Comment by ZhouyihaiDing** [comment]
It can only be reproduced on gLinux Debian. Will it related to something like openssl version?
I get the log here: https://gist.github.com/ZhouyihaiDing/3bb88ed29a5e73d7af916efdcf77a4d3

**Comment by jtattermusch** [comment]
similar to https://github.com/grpc/grpc/issues/9538.

**Comment by jtattermusch** [comment]
@ZhouyihaiDing   is there a way how to reproduce this in C/C++ (outside of PHP)?

**Comment by ZhouyihaiDing** [comment]
Okay, I'll try to reproduce it with C++.
It may caused by macro or initiate value difference between different systems, although I haven't verify it. Otherwise it's strange it works fine in ubuntu 16,17 but failed in debian 4.9.65.

**Comment by ZhouyihaiDing** [comment]
It's duplicated with #15582. And the `TIMEOUT` part should be solved by #15594.
